README.md

```diff
@@ -1,142 +1,220 @@
+developing WSO2 EI integration artifacts with enables to structure the whole enterprise integration solution to a 
+single project and export it as a single deployable unit. A team can push the project on Github and collaboratively 
+work on developing integration flows. 
 
-Repository URL <https://github.com/wso2/wso2-commons-vfs>  
-This contains the forked apache commons vfs library. Provides file processing capabilities to ESB.  
-Dependent repository:wso2-syanspe
+Read on how to develop integration artifacts with EI tooling [here](https://docs.wso2.com/display/EI640/Integration+Tutorials).
 
-#### wso2 synapse
+## Building from Source
 
-Repository URL <https://github.com/wso2/wso2-synapse>  
-Provides most of the ESB functionality to WSO2 Enterprise Service Bus.  
-Dependent repository:carbon-mediation
+If you intend to build the project from the source you can do that with or without building the dependent projects. 
+Here is an outline how the dependent project are structured. If you build with dependencies, you need to do it from 
+bottom to top in the hierarchy.
 
-#### carbon-mediation
+![WSO2 EI-Repositories](gh-docs/images/ei-repo-structure.png?raw=true)
 
-Repository URL <https://github.com/wso2/carbon-mediation>  
-Implements more functionality on top of wso2-synapse.  
-Dependent repository:product-ei
+Repositories referred above
 
-#### carbon-data
+- [product EI](https://github.com/wso2/product-ei): contains product EI packaging 
+- [carbon-mediation](https://github.com/wso2/carbon-mediation): mediation features related to WSO2 ESB are developed in this repository 
+- [carbon-data](https://github.com/wso2/carbon-data): contains features related to data services 
+- [wso2-synapse](https://github.com/wso2/wso2-synapse): source of messaging engine used by the WSO2 ESB runtime 
+- [carbon-business-messaging](https://github.com/wso2/carbon-business-messaging): features related to WSO2 Message Broker 
+- [andes](https://github.com/wso2/andes): implementation of messaging core of WSO2 Message Broker 
+- [carbon-business-process](https://github.com/wso2/carbon-business-process): contains the modules implementing BPEL , WS-Human Tasks and BPMN support for WSO2 BPS 
+- [wso2-ode](https://github.com/wso2/wso2-ode): WS-BPEL compliant web services orchestration engine 
+- [carbon Commons](https://github.com/wso2/carbon-commons): contains the common components to Carbon (i.e cluster mgt, logging, ntask) 
+- [WSO2 Axis2 Transports](https://github.com/wso2/wso2-axis2-transports): includes the axis2 based transport implementations (i.e JMS, Mail, message builders, formatters) 
+- [carbon Kernel (4.x.x)](https://github.com/wso2/carbon-kernel/tree/4.4.x): lean, modular, OSGi-based platform, is the base of the WSO2 Carbon platform. server software development platform 
+- [wso2-axis2](https://github.com/wso2/wso2-axis2): is a Web Services / SOAP / WSDL engine. Forked by [Apache Axis2](http://axis.apache.org/axis2/java/core/) 
```

Comment:
Is it "forked by" or "forked from"? 

"Forked by" indicates that they have forked our repo. I assume that it's the opposite?

Reply:
Should be "forked from". Updated.